File: 백준/Bronze/1009. 분산처리/분산처리.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    private static long power(long base, long exp, long mod) {
        long result = 1;
        base %= mod;

        while (exp > 0) {
            if (exp % 2 == 1) {
                result = (result * base) % mod;
            }

            base = (base * base) % mod;
            exp /= 2;
        }

        return result;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());

        StringBuilder sb = new StringBuilder();
        while (t-- > 0) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            long a = Long.parseLong(st.nextToken());
            long b = Long.parseLong(st.nextToken());

            long result = power(a, b, 10);

            if (result == 0) result = 10;

            sb.append(result).append('\n');
        }

        System.out.print(sb);
    }
}

File: 백준/Bronze/1009. 분산처리/분산처리.kt
import java.io.BufferedReader
import java.io.InputStreamReader

private fun power(base: Long, exp: Long, mod: Long): Long {
    var result = 1L
    var b = base % mod
    var e = exp

    while (e > 0) {
        if (e % 2 == 1L) {
            result = (result * b) % mod
        }

        b = (b * b) % mod
        e /= 2
    }

    return result
}

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val t = br.readLine().toInt()

    val sb = StringBuilder()
    repeat(t) {
        val (a, b) = br.readLine().split(" ").map { it.toLong() }
        var result = power(a, b, 10)

        if (result == 0L) result = 10L

        sb.append(result).append('\n')
    }

    print(sb)
}
